go/config/config_test.go

```diff
@@ -0,0 +1,301 @@
+	}
+	defer func() {
+		_ = os.Remove(file.Name())
+	}()
+
+	_, err = file.Write(userConfig_memstore_yaml(t))
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	err = file.Close()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	cfg, err := LoadConfig(file.Name())
+	if err != nil {
+		t.Error(err)
+	}
+
+	if cfg.API == nil {
+		t.Errorf("config API is nil")
+	}
+
+	if !cmp.Equal(cfg.API, userServerConfig(t)) {
+		t.Errorf("Values in cfg.API are not correct\n%s", cmp.Diff(cfg.API, userServerConfig(t)))
+	}
+
+	if cfg.StorageType != "memstore" {
+		t.Errorf("Storage type %s is not memstore", cfg.StorageType)
+	}
+
+	if cfg.StorageConfig != nil {
+		t.Errorf("storage configuration is not nil")
+	}
+}
+
+func TestLoadConfig_ReturnsConfig_UserSuppliedValues_Embedded(t *testing.T) {
+	file, err := ioutil.TempFile("", "config.*.yaml")
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer func() {
+		_ = os.Remove(file.Name())
+	}()
+
+	_, err = file.Write(userConfig_embedded_yaml(t))
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	err = file.Close()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	cfg, err := LoadConfig(file.Name())
+	if err != nil {
+		t.Error(err)
+	}
+
+	if cfg.API == nil {
+		t.Errorf("config API is nil")
+	}
+
+	if !cmp.Equal(cfg.API, userServerConfig(t)) {
+		t.Errorf("Values in cfg.API are not correct\n%s", cmp.Diff(cfg.API, userServerConfig(t)))
+	}
+
+	if cfg.StorageType != "embedded" {
+		t.Errorf("Storage type %s is not embedded", cfg.StorageType)
+	}
+
+	if cfg.StorageConfig == nil {
+		t.Errorf("storage configuration is nil")
+	}
+
+	var storeConfig EmbeddedStoreConfig
+
+	err = ConvertGenericConfigToSpecificType(*cfg.StorageConfig, &storeConfig)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	if !cmp.Equal(storeConfig, *userEmbeddedConfig(t)) {
+		t.Errorf("Values in storage configuration are not correct\n%s", cmp.Diff(storeConfig, *userEmbeddedConfig(t)))
+	}
+}
+
+// TODO(#341) move these 2 supporting functions and the test case to the new project
+func userPostgresConfig(t *testing.T) *PgSQLConfig {
+	return &PgSQLConfig{
+		Host:          "127.0.0.1:5432",
+		DbName:        "postgres",
+		User:          "postgres",
+		Password:      "password",
+		SSLMode:       "require",
+		PaginationKey: "",
+	}
+}
+
+func userConfig_postgres_yaml(t *testing.T) []byte {
+	return []byte(`
+grafeas:
+  api:
+    address: "0.0.0.0:8081"
+    certfile: abc
+    keyfile: def
+    cafile:  ghi
+    cors_allowed_origins:
+      - "http://example.com"
+      - "https://somewhere.else.com"
+  storage_type: "postgres"
+  postgres:
+    host: "127.0.0.1:5432"
+    dbname: "postgres"
+    user: "postgres"
+    password: "password"
+    sslmode: "require"
+    paginationkey:
+`)
+}
+
+func TestLoadConfig_ReturnsConfig_UserSuppliedValues_Postgres(t *testing.T) {
+	file, err := ioutil.TempFile("", "config.*.yaml")
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer func() {
+		_ = os.Remove(file.Name())
+	}()
+
+	_, err = file.Write(userConfig_postgres_yaml(t))
+	if err != nil {
```

Comment:
Try not to reuse error vars and it can obfuscate the err if there's a coding bug. Instead, inline the if check for lines like this (and everywhere else). E.g.,

if _, err := file.Write(...); if err != nil {
  t.Errorf(...)
}

Reply:
Amended.